src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/UpdateFormStoreToolbarAction.js

```diff
@@ -0,0 +1,309 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import React from 'react';
+import symfonyRouting from 'fos-jsrouting/router';
+import jexl from 'jexl';
+import Dialog from '../../../components/Dialog';
+import {Requester} from '../../../services';
+import {translate} from '../../../utils';
+import FormContainer, {memoryFormStoreFactory} from '../../../containers/Form';
+import Router from '../../../services/Router';
+import Form from '../Form';
+import {ResourceStore} from '../../../stores';
+import AbstractFormToolbarAction from './AbstractFormToolbarAction';
+import type {ResourceFormStore} from '../../../containers';
+import type {FormStoreInterface} from '../../../containers/Form';
+
+export default class UpdateFormStoreToolbarAction extends AbstractFormToolbarAction {
+    @observable loading = false;
+    @observable showDialog = false;
+    formStore: FormStoreInterface;
+
+    constructor(
+        resourceFormStore: ResourceFormStore,
+        form: Form,
+        router: Router,
+        locales: ?Array<string>,
+        options: { [key: string]: mixed },
+        parentResourceStore: ResourceStore
+    ) {
+        super(
+            resourceFormStore,
+            form,
+            router,
+            locales,
+            options,
+            parentResourceStore
+        );
+
+        // Required options validation
+        const requiredOptions = [
+            'icon',
+            'route',
+            'contentExpressions',
+            'dialogKey',
+            'dialogTitle',
+            'dialogDescription',
+        ];
+
+        const missingOptions = requiredOptions.filter((key) => !options[key]);
+        if (missingOptions.length > 0) {
+            throw new Error(`Missing required options: ${missingOptions.join(', ')}`);
+        }
+
+        // Validate content expressions
+        if (!Array.isArray(options.contentExpressions)) {
+            throw new Error('contentExpressions must be an array of objects with get and path properties');
+        }
+    }
+
+    @computed get label() {
```

Comment:
strange usage of `@computed` but seems like we did already do this in Form.js so fine.